js/src/util/index.js

```diff
@@ -36,7 +36,19 @@ const getSelector = element => {
   let selector = element.getAttribute('data-bs-target')
 
   if (!selector || selector === '#') {
-    const hrefAttr = element.getAttribute('href')
+    let hrefAttr = element.getAttribute('href')
+
+    // The only valid content that could double as a selector are ids, so everything starting with . or #
+    // If a 'real' url is used as the selector, document.querySelector will rightfully
+    // complain it is invalid. (see twbs#32273)
+    if (!hrefAttr || (!hrefAttr.includes('#') && !hrefAttr.startsWith('.'))) {
```

Comment:
It restricts the user to use only the class and id selectors in the href attribute. This restriction was not before :thinking:

Reply:
I do not see a problem there. I did not find any mention in the documentation, that anything other than IDs in the href are even supported.

Considering that that freedom for using invalid hrefs as selectors was exactly what caused problems and lead to this PR (and this is the big jump to v5), I think this would be the last opportunity to make such a change before possibly a myriad of users starts misusing the href in that way.

Besides, that's what the `data-bs-target` attribute is for, right? (which is encouraged everywhere in the docs)

Reply:
Actually, this would fix the issue we have. I guess it's a good solution for v5. We can revisit in v6.